src/rust/src/x509.rs

```diff
@@ -43,6 +43,505 @@ lazy_static::lazy_static! {
+                    Ok(Some(parse_authority_key_identifier(py, ext_data)?))
+                } else if oid == &*CRL_DISTRIBUTION_POINTS_OID {
+                    let dp = parse_distribution_points(py, ext_data)?;
+                    Ok(Some(
+                        x509_module.getattr("CRLDistributionPoints")?.call1((dp,))?,
+                    ))
+                } else if oid == &*FRESHEST_CRL_OID {
+                    let dp = parse_distribution_points(py, ext_data)?;
+                    Ok(Some(x509_module.getattr("FreshestCRL")?.call1((dp,))?))
+                } else if oid == &*PRECERT_SIGNED_CERTIFICATE_TIMESTAMPS_OID {
+                    let contents = asn1::parse_single::<&[u8]>(ext_data)?;
+                    let scts = parse_scts(py, contents, LogEntryType::PreCertificate)?;
+                    Ok(Some(
+                        x509_module
+                            .getattr("PrecertificateSignedCertificateTimestamps")?
+                            .call1((scts,))?,
+                    ))
+                } else if oid == &*NAME_CONSTRAINTS_OID {
+                    let nc = asn1::parse_single::<NameConstraints<'_>>(ext_data)?;
+                    let permitted_subtrees = match nc.permitted_subtrees {
+                        Some(data) => parse_general_subtrees(py, data)?,
+                        None => py.None(),
+                    };
+                    let excluded_subtrees = match nc.excluded_subtrees {
+                        Some(data) => parse_general_subtrees(py, data)?,
+                        None => py.None(),
+                    };
+                    Ok(Some(
+                        x509_module
+                            .getattr("NameConstraints")?
+                            .call1((permitted_subtrees, excluded_subtrees))?,
+                    ))
+                } else {
+                    Ok(None)
+                }
+            },
+        )
+    }
+    // This getter exists for compatibility with pyOpenSSL and will be removed.
+    // DO NOT RELY ON IT. WE WILL BREAK YOU WHEN WE FEEL LIKE IT.
+    #[getter]
+    fn _x509<'p>(
+        slf: pyo3::pycell::PyRef<'_, Self>,
+        py: pyo3::Python<'p>,
+    ) -> Result<&'p pyo3::PyAny, PyAsn1Error> {
+        let cryptography_warning = py
+            .import("cryptography.utils")?
+            .getattr("CryptographyDeprecationWarning")?;
+        let warnings = py.import("warnings")?;
+        warnings.call_method1(
+            "warn",
+            (
+                "This version of cryptography contains a temporary pyOpenSSL fallback path. Upgrade pyOpenSSL now.",
+                cryptography_warning,
+            ),
+        )?;
+        let backend = py
+            .import("cryptography.hazmat.backends.openssl.backend")?
+            .getattr("backend")?;
+        Ok(backend.call_method1("_cert2ossl", (slf,))?)
+    }
+}
+
+fn cert_version(py: pyo3::Python<'_>, version: u8) -> Result<&'_ pyo3::PyAny, PyAsn1Error> {
```

Comment:
```suggestion
fn cert_version(py: pyo3::Python<'_>, version: u8) -> Result<&pyo3::PyAny, PyAsn1Error> {
```

Should be sufficient?